src/analysis/calculus/mean_value.lean

```diff
@@ -922,57 +922,60 @@ theorem strict_mono_of_deriv_pos {f : ℝ → ℝ} (hf : differentiable ℝ f)
 of the real line. If `f` is differentiable on the interior of `D` and `f'` is negative, then
 `f` is a strictly antitone function on `D`. -/
-theorem convex.strict_antitone_of_deriv_neg {D : set ℝ} (hD : convex ℝ D) {f : ℝ → ℝ}
+theorem convex.strict_anti_on_of_deriv_neg {D : set ℝ} (hD : convex ℝ D) {f : ℝ → ℝ}
   (hf : continuous_on f D) (hf' : differentiable_on ℝ f (interior D))
   (hf'_neg : ∀ x ∈ interior D, deriv f x < 0) :
-  ∀ x y ∈ D, x < y → f y < f x :=
-by simpa only [zero_mul, sub_lt_zero] using hD.image_sub_lt_mul_sub_of_deriv_lt hf hf' hf'_neg
+  strict_anti_on f D :=
+λ x hx y, by simpa only [zero_mul, sub_lt_zero]
+  using hD.image_sub_lt_mul_sub_of_deriv_lt hf hf' hf'_neg x y hx
 
 /-- Let `f : ℝ → ℝ` be a differentiable function. If `f'` is negative, then
 `f` is a strictly antitone function. -/
-theorem strict_antitone_of_deriv_neg {f : ℝ → ℝ} (hf : differentiable ℝ f)
+theorem strict_anti_of_deriv_neg {f : ℝ → ℝ} (hf : differentiable ℝ f)
   (hf' : ∀ x, deriv f x < 0) :
-  ∀ ⦃x y⦄, x < y → f y < f x :=
-λ x y hxy, convex_univ.strict_antitone_of_deriv_neg hf.continuous.continuous_on hf.differentiable_on
-  (λ x _, hf' x) x y trivial trivial hxy
+  strict_anti f :=
+λ x y hxy, convex_univ.strict_anti_on_of_deriv_neg hf.continuous.continuous_on hf.differentiable_on
+  (λ x _, hf' x) trivial trivial hxy
 
 /-- Let `f` be a function continuous on a convex (or, equivalently, connected) subset `D`
 of the real line. If `f` is differentiable on the interior of `D` and `f'` is nonpositive, then
 `f` is an antitone function on `D`. -/
-theorem convex.antitone_of_deriv_nonpos {D : set ℝ} (hD : convex ℝ D) {f : ℝ → ℝ}
+theorem convex.antitone_on_of_deriv_nonpos {D : set ℝ} (hD : convex ℝ D) {f : ℝ → ℝ}
   (hf : continuous_on f D) (hf' : differentiable_on ℝ f (interior D))
   (hf'_nonpos : ∀ x ∈ interior D, deriv f x ≤ 0) :
-  ∀ x y ∈ D, x ≤ y → f y ≤ f x :=
-by simpa only [zero_mul, sub_nonpos] using hD.image_sub_le_mul_sub_of_deriv_le hf hf' hf'_nonpos
+  antitone_on f D :=
+λ x hx y hy hxy, by simpa only [zero_mul, sub_nonpos]
+  using hD.image_sub_le_mul_sub_of_deriv_le hf hf' hf'_nonpos x y hx hy hxy
 
 /-- Let `f : ℝ → ℝ` be a differentiable function. If `f'` is nonpositive, then
 `f` is an antitone function. -/
 theorem antitone_of_deriv_nonpos {f : ℝ → ℝ} (hf : differentiable ℝ f) (hf' : ∀ x, deriv f x ≤ 0) :
-  ∀ ⦃x y⦄, x ≤ y → f y ≤ f x :=
-λ x y hxy, convex_univ.antitone_of_deriv_nonpos hf.continuous.continuous_on hf.differentiable_on
-  (λ x _, hf' x) x y trivial trivial hxy
+  antitone f :=
+λ x y hxy, convex_univ.antitone_on_of_deriv_nonpos hf.continuous.continuous_on hf.differentiable_on
+  (λ x _, hf' x) trivial trivial hxy
 
 /-- If a function `f` is continuous on a convex set `D ⊆ ℝ`, is differentiable on its interior,
 and `f'` is monotone on the interior, then `f` is convex on `D`. -/
 theorem convex_on_of_deriv_mono {D : set ℝ} (hD : convex ℝ D) {f : ℝ → ℝ}
```

Comment:
```suggestion
theorem convex_on_of_deriv_monotone_on {D : set ℝ} (hD : convex ℝ D) {f : ℝ → ℝ}
```
or if you don't think `on` is important,
```suggestion
theorem convex_of_deriv_mono {D : set ℝ} (hD : convex ℝ D) {f : ℝ → ℝ}
```

Reply:
The two `_on` have quite a different statuses in my opinion because `monotone` and `monotone_on` are much more closely related than `convex` and `convex_on`. That being said, I think both are important.

Reply:
I'll let you make the call, just checking you hadn't overlooked this one